src/DbInterface.cpp

```diff
@@ -584,6 +584,47 @@ void DbInterface::getServerIpAddress(std::shared_ptr<swss::DBConnector> configDb
     processServerIpAddress(entries);
 }
 
+//
+// ---> processPortCableType(std::vector<swss::KeyOpFieldsValuesTuple> &entries);
+//
+// process port cable type and build a map of port name to cable type
+//
+void DbInterface::processPortCableType(std::vector<swss::KeyOpFieldsValuesTuple> &entries)
+{
+    for (auto &entry: entries) {
+        std::string portName = kfvKey(entry);
+        std::string operation = kfvOp(entry);
+        std::vector<swss::FieldValueTuple> fieldValues = kfvFieldsValues(entry);
+
+        std::string field = "cable_type";
+        std::vector<swss::FieldValueTuple>::const_iterator cit = std::find_if(
+            fieldValues.cbegin(),
+            fieldValues.cend(),
+            [&field] (const swss::FieldValueTuple &fv) {return fvField(fv) == field;}
+        );
+        std::string portCableType = (cit != fieldValues.cend() ? cit->second : "active-standby");
```

Comment:
So if ```cable_type``` is missing,  the default is ```active-standby```. It's accurate for now, but moving forward to active-active, should the default be ```active-standby```?

Reply:
For the future `active-active`, if we can guarantee that the `cable_type` is always set for entries in `MUX_CABLE`, we are good here, right?